lapis2/src/main/kotlin/org/genspectrum/lapis/scheduler/DataVersionCacheInvalidator.kt

```diff
@@ -0,0 +1,44 @@
+package org.genspectrum.lapis.scheduler
+
+import mu.KotlinLogging
+import org.genspectrum.lapis.silo.CachedSiloClient
+import org.genspectrum.lapis.silo.SILO_QUERY_CACHE_NAME
+import org.springframework.cache.annotation.CacheEvict
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.stereotype.Component
+import java.util.concurrent.TimeUnit
+
+private val log = KotlinLogging.logger {}
+
+@Component
+class DataVersionCacheInvalidator(
+    private val cachedSiloClient: CachedSiloClient,
+    private val cacheClearer: CacheClearer,
+) {
+    final var currentlyCachedDataVersion = "uninitialized"
+        private set
+
+    @Scheduled(fixedRate = 30, timeUnit = TimeUnit.SECONDS)
```

Comment:
@chaoran-chen What's a reasonable interval to check for new versions? I think the old LAPIS does it every second, if I interpret this correctly:
https://github.com/GenSpectrum/LAPIS/blob/831399692fdf1bcce6fc9e4021e950c4ee8bc00f/server/src/main/java/ch/ethz/lapis/api/DataVersionService.java#L40

Reply:
Yes, every second would be good.

Reply:
Ok, then I can probably also remove the other logic that checks the data version on a normal query.